### DISCUSSION TITLE
Doubt: How to implement plus icon on hover?

### DISCUSSION THREAD

**Discussion by saksham9312** [microsoft/vscode-discussions#1057] [Extension Development QnA] [answered]
We are building an extension in which we require to get line number in a active text file. For this we need a plus kind of icon in left ruler. This plus icon should appear on hover when we hover the mouse near the line number. Otherwise it should not appear.

As shown in the image. Can someone give us code or implementation for it? We are new to VSCode development.
<img width="722" alt="img4" src="https://github.com/microsoft/vscode-discussions/assets/63234460/3ecb7fe6-c541-406d-9d54-09ae7dd6556e">

**Comment by alefragnani** [comment, marked as answer]
Hi @saksham9312 ,

There is no API available to replicate the exact behavior your are describing. This ➕ icon you see while hovering the gutter/line number is part of the `Comments API` (https://github.com/microsoft/vscode-extension-samples/tree/main/comment-sample).

You could, however, look for other APIs that works _while hovering_ like `HoveProvider` (https://code.visualstudio.com/api/language-extensions/programmatic-language-features#show-hovers), `Code Actions` (https://code.visualstudio.com/api/language-extensions/programmatic-language-features#possible-actions-on-errors-or-warnings) or `CodeLens` (https://code.visualstudio.com/api/language-extensions/programmatic-language-features#codelens-show-actionable-context-information-within-source-code`. Maybe `Code Actions` would be better fit, since it not only displays something about some word in a line, but also allows you to run some command on it.

Another alternative would be to use the recent `editor/lineNumber/context` contribution point (https://code.visualstudio.com/updates/v1_78#_editorlinenumbercontext-menu), where you can add menu items to the context menu in the gutter/line number, like:

![](https://user-images.githubusercontent.com/3781424/224760895-acd2e460-42a4-4eb7-b7be-f6930bf40315.png)

Hope this helps

**Comment by theReynald** [comment]
@saksham9312, feel free to reopen if you need more info. Closing this thread for now.
